web/src/lib/components/memory-page/memory-viewer.svelte

```diff
@@ -0,0 +1,354 @@
+<script lang="ts">
+	import { memoryStore } from '$lib/stores/memory.store';
+	import { DateTime } from 'luxon';
+	import { onMount } from 'svelte';
+	import { MemoryLaneResponseDto, api } from '@api';
+	import { goto } from '$app/navigation';
+	import ControlAppBar from '$lib/components/shared-components/control-app-bar.svelte';
+	import Play from 'svelte-material-icons/Play.svelte';
+	import Pause from 'svelte-material-icons/Pause.svelte';
+	import ChevronDown from 'svelte-material-icons/ChevronDown.svelte';
+	import ChevronUp from 'svelte-material-icons/ChevronUp.svelte';
+	import ChevronLeft from 'svelte-material-icons/ChevronLeft.svelte';
+	import ChevronRight from 'svelte-material-icons/ChevronRight.svelte';
+	import { AppRoute } from '$lib/constants';
+	import { page } from '$app/stores';
+	import noThumbnailUrl from '$lib/assets/no-thumbnail.png';
+	import GalleryViewer from '$lib/components/shared-components/gallery-viewer/gallery-viewer.svelte';
+	import CircleIconButton from '$lib/components/elements/buttons/circle-icon-button.svelte';
+	import IntersectionObserver from '$lib/components/asset-viewer/intersection-observer.svelte';
+	import { fade } from 'svelte/transition';
+
+	let currentIndex = 0;
+	let currentMemory: MemoryLaneResponseDto;
+	let nextMemory: MemoryLaneResponseDto;
+	let lastMemory: MemoryLaneResponseDto;
+
+	let lastIndex = 0;
+	let nextIndex = 0;
+	$: showNextMemory = nextIndex <= $memoryStore?.length - 1;
+	$: showPreviousMemory = currentIndex != 0;
+
+	let memoryGallery: HTMLElement;
+	let memoryWrapper: HTMLElement;
+	let galleryInView = false;
+
+	onMount(async () => {
+		if (!$memoryStore) {
+			const timezone = DateTime.local().zoneName;
+			const { data } = await api.assetApi.getMemoryLane({ timezone });
+			$memoryStore = data;
+		}
+
+		const queryIndex = $page.url.searchParams.get('index');
+		if (queryIndex != null) {
+			currentIndex = parseInt(queryIndex);
+			if (isNaN(currentIndex) || currentIndex > $memoryStore.length - 1) {
+				currentIndex = 0;
+			}
+		}
+
+		currentMemory = $memoryStore[currentIndex];
+
+		nextIndex = currentIndex + 1;
+		nextMemory = $memoryStore[nextIndex];
+
+		if (currentIndex > 0) {
+			lastMemory = $memoryStore[lastIndex];
+		}
+	});
+
+	const toNextMemory = (): boolean => {
+		if (showNextMemory) {
+			resetAutoPlay();
+
+			currentIndex += 1;
+			nextIndex = currentIndex + 1;
+			lastIndex = currentIndex - 1;
+
+			currentMemory = $memoryStore[currentIndex];
+			nextMemory = $memoryStore[nextIndex];
+			lastMemory = $memoryStore[lastIndex];
+
+			return true;
+		}
+
+		return false;
```

Comment:
Given the condition checks against `showNextMemory`, it may be simpler just to return that value.

```suggestion
		return showNextMemory;
```